src/webgpu/shader/validation/parse/statement_behavior.spec.ts

```diff
@@ -0,0 +1,143 @@
+  loop4: `loop { continuing { break; } }`,
+  loop5: `loop { continuing { continue; } }`,
+  loop6: `loop { continuing { return; } }`,
+  loop7: `loop { continue; break; }`,
+  loop8: `loop { continuing { break if true; return; } }`,
+  for1: `for (;;) { }`,
+  for2: `for (var i = 0; ; i++) { }`,
+  for3: `for (;; break) { }`,
+  for4: `for (;; continue ) { }`,
+  for5: `for (;; return ) { }`,
+  for6: `for (;;) { continue; break; }`,
+  // while loops always have break in their behaviors.
+  switch1: `switch (1) { case 1 { } }`,
+  sequence1: `return; loop { }`,
+  compound1: `{ loop { } }`,
+};
+
+g.test('invalid_statements')
+  .desc('Test statements with invalid behaviors')
+  .specURL('https://gpuweb.github.io/gpuweb/wgsl/#behaviors-rules')
+  .params(u => u.combine('body', keysOf(kInvalidStatements)))
+  .fn(t => {
+    const body = kInvalidStatements[t.params.body];
+    const code = `fn foo() {
+      ${body};
+    }`;
+    t.expectCompileResult(false, code);
+  });
+
+const kValidStatements = {
+  empty: ``,
+  const_assert: `const_assert true`,
+  let: `let x = 1`,
+  var1: `var x = 1`,
+  var2: `var x : i32`,
+  assign: `v = 1`,
+  phony_assign: `_ = 1`,
+  compound_assign: `v += 1`,
+  return: `return`,
+  discard: `discard`,
+  function_call1: `bar()`,
+  function_call2: `workgroupBarrier()`,
+
+  if1: `if true { } else { }`,
+  if2: `if true { }`,
+
+  break1: `loop { break; }`,
+  break2: `loop { if false { break; } }`,
+  break_if: `loop { continuing { break if false; } }`,
+
+  continue1: `loop { continue; continuing { break if true; } }`,
+
+  loop1: `loop { break; }`,
+  loop2: `loop { break; continuing { } }`,
+  loop3: `loop { continue; continuing { break if true; } }`,
+  loop4: `loop { break; continue; }`,
+
+  for1: `for (; true; ) { }`,
+  for2: `for (;;) { break; }`,
+  for3: `for (;true;) { continue; }`,
+
+  while1: `while true { }`,
+  while2: `while true { continue; }`,
+  while3: `while true { continue; break; }`,
+
+  switch1: `switch 1 { default { } }`,
+  swtich2: `switch 1 { case 1 { } default { } }`,
+  switch3: `switch 1 { default { break; } }`,
+  switch4: `switch 1 { default { } case 1 { break; } }`,
+
+  sequence1: `return; let x = 1`,
+  sequence2: `if true { } let x = 1`,
+  sequence3: `switch 1 { default { break; return; } }`,
+
+  compound1: `{ }`,
+  compound2: `{ loop { break; } if true { return; } }`,
+};
+
+g.test('valid_statements')
+  .desc('Test statements with valid behaviors')
+  .specURL('https://gpuweb.github.io/gpuweb/wgsl/#behaviors-rules')
+  .params(u => u.combine('body', keysOf(kValidStatements)))
+  .fn(t => {
+    const body = kValidStatements[t.params.body];
+    const code = `
+    var<private> v : i32;
+    fn bar() { }
+    fn foo() {
+      ${body};
+    }`;
+    t.expectCompileResult(true, code);
+  });
+
+const kInvalidFunctions = {
+  next_for_type: `fn foo() -> bool { }`,
+  next_return_for_type: `fn foo() -> bool { if true { return true } }`,
```

Comment:
Missing semicolon on the return statement, might cause it to be rejected for the wrong reasons.